server/src/test/java/io/spine/server/ServerEnvironmentTest.java

```diff
@@ -205,6 +210,67 @@ void testsFactory() {
         }
     }
 
+    @Nested
+    @DisplayName("configure `StorageFactory` for tests")
+    class TestStorageFactoryConfig {
+
+        @AfterEach
+        void resetEnvironment() {
+            serverEnvironment.reset();
+        }
+
+        @Test
+        @DisplayName("returning it when explicitly set")
+        void getSet() {
+            StorageFactory factory = new StubStorageFactory();
+
+            serverEnvironment.configureStorageForTests(factory);
+            assertThat(serverEnvironment.storageFactory())
+                    .isEqualTo(factory);
+        }
+    }
+
+    @Nested
+    @DisplayName("configure `TransportFactory`")
+    class TransportFactoryConfig {
+
+        private final Environment environment = Environment.instance();
+
+        @BeforeEach
+        void turnToProduction() {
+            environment.setToProduction();
+        }
+
+        @AfterEach
+        void backToTests() {
+            environment.setToTests();
+            serverEnvironment.reset();
+        }
+
+        @Test
+        @DisplayName("throw NPE if not configured")
+        void throwsIfNotConfigured() {
+            assertThrows(NullPointerException.class, serverEnvironment::transportFactory);
+        }
+
+        @Test
+        @DisplayName("do not allow setting `InMemoryTransportFactory`")
+        void noInMemProdTransport() {
```

Comment:
Actually, we may use in-memory transport in production. That works flawlessly if the bounded contexts are deployed on the same node.

Let's discuss in person.